crates/miden-testing/src/kernel_tests/tx/test_tx.rs

```diff
@@ -1396,8 +1400,8 @@ async fn execute_tx_view_script() -> anyhow::Result<()> {
     ";
 
     let source = NamedSource::new("test::module_1", test_module_source);
-    let source_manager = Arc::new(DefaultSourceManager::default());
-    let assembler = TransactionKernel::assembler();
+    let source_manager = default_source_manager_arc_dyn();
+    let assembler = TransactionKernel::assembler_with_source_manager(source_manager.clone());
```

Comment:
This looks correct, but for some reason, inserting a `push.3 mem_loadw` into the script used here does result in an error, but does not include the source snippet. For reference, I ran `BUILD_GENERATED_FILES_IN_SRC=1 RUST_BACKTRACE=1 cargo nextest run --profile default --cargo-profile test-dev --features concurrent,testing,std --features with-debug-info --no-capture execute_tx_view_script`. The issue may be the missing `with_debug_mode(true)` on the assembler, but enabling it results in this panic (https://github.com/0xMiden/miden-vm/issues/1972), when compiling the tx script, but interestingly, the std library is linked dynamically here.

Not an actionable comment for this PR, I think, but wanted to mention it.